rules/docker-nat.mk

```diff
@@ -0,0 +1,30 @@
+# docker image for nat
+
+DOCKER_NAT_STEM = docker-nat
+DOCKER_NAT = $(DOCKER_NAT_STEM).gz
+DOCKER_NAT_DBG = $(DOCKER_NAT_STEM)-$(DBG_IMAGE_MARK).gz
+
+$(DOCKER_NAT)_PATH = $(DOCKERS_PATH)/$(DOCKER_NAT_STEM)
+
+$(DOCKER_NAT)_DEPENDS += $(SWSS) $(REDIS_TOOLS) $(IPTABLESIP4TC) $(IPTABLESIP6TC) $(IPTABLESIPTC) $(IPXTABLES12) $(IPTABLES) 
+$(DOCKER_NAT)_DBG_DEPENDS = $($(DOCKER_CONFIG_ENGINE_STRETCH)_DBG_DEPENDS)
+$(DOCKER_NAT)_DBG_DEPENDS += $(SWSS_DBG) $(LIBSWSSCOMMON_DBG)
+$(DOCKER_NAT)_DBG_IMAGE_PACKAGES = $($(DOCKER_CONFIG_ENGINE_STRETCH)_DBG_IMAGE_PACKAGES)
+
+$(DOCKER_NAT)_LOAD_DOCKERS += $(DOCKER_CONFIG_ENGINE_STRETCH)
+
+SONIC_DOCKER_IMAGES += $(DOCKER_NAT)
+SONIC_INSTALL_DOCKER_IMAGES += $(DOCKER_NAT)
```

Comment:
This means we have nat docker by default for every platform?
Do we have a way to disable it for platforms that do not have NAT implemented yet?

Reply:
@stepanblyschak Yes, NAT docker will be started by default on every platform. But NAT feature be admin mode disabled by default. Can we disable a service based on the platform checks on bootup? Is it being done for any service already in SONiC?

Reply:
I don't think we have a way to decide this on bootup.
One useful thing to have would be a build flag 'ENABLE_NAT' in the same way as ENABLE_TELEMETRY or ENABLE_SFLOW.
ENABLE_NAT=y by default is ok if there is a check if nat feature is supported in orchagent.

Reply:
@stepanblyschak Marian was mentioning, instead of disabling during compile time, we have to prevent/disable it run time in OA for platforms that have not implemented the feature. That is taken care of.

Reply:
Ok